providers/os/provider/provider_test.go

```diff
@@ -92,7 +90,7 @@ func TestLocalConnectionIdDetectors_DelayedDiscovery(t *testing.T) {
 	require.Contains(t, connectResp.Asset.IdDetector, ids.IdDetector_CloudDetect)
 	require.NotContains(t, connectResp.Asset.IdDetector, ids.IdDetector_SshHostkey)
 	require.Len(t, connectResp.Asset.PlatformIds, 1)
-	require.Nil(t, connectResp.Asset.Platform)
+	require.Nil(t, connectResp.Asset.Platform) // Why does this need to be nil
```

Comment:
I don't know why this needs to be nil

Reply:
if we have the `DelayDiscovery` set to true, then the platform is not set because we haven't performed the "discover" step